### PR TITLE
TPSVC-16642: [LOGGING] [DESIGN] Design ECS format integration in Daikon Logging

### PR DESCRIPTION
**Pull request by jehervy**
**What is the problem this Pull Request is trying to solve?**
As ECS (Elastic Common Schema) has been chosen to become Talend common logging format, its integration must be designed.
 
**What is the chosen solution to this problem?**
Based on requirements, proposals must be done in order to facilitate ECS adoption.
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-logging/doc/ECS_migration.adoc [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+* Easing analyse and visualize
+* Reducing storage size
+
+Moreover, this format must be implemented and maintained at one place in order to facilitate the future evolutions.
+
+=== Adoption
+
+One of the main requirements is the ease of adoption. The goal is that each team uses the same format, this goal will be reached only if the effort to move to this format is minimal :
+
+* The migration must not break the current format
```
Comment (sgandon) [inline daikon-logging/doc/ECS_migration.adoc:31]: this is not clear ! what format are you talking about here ?

File: daikon-logging/doc/ECS_migration.adoc [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+* The expected format must be clear and shouldn't change with time
+* The cost of the migration should be minimal
+* The Backward compatibility must be preserved (in order to adapt dashboards to new format)
```
Comment (sgandon) [inline daikon-logging/doc/ECS_migration.adoc:34]: backward compatibility of what the output format or the APIs ?
If it is about output format, then what does it mean exactly ?

File: daikon-logging/doc/ECS_migration.adoc [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+* It must be easy for each team to extend the format by adding custom fields
```
Comment (sgandon) [inline daikon-logging/doc/ECS_migration.adoc:35]: as long at it follows the ECS standard even for custom fields.

File: daikon-logging/doc/ECS_migration.adoc [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+=== Troubleshooting
+
+One of the main function of a log being to help the troubleshooting, moving to this new format should facilitate it by :
+
+* Categorizing the logs
+* Formalizing, documenting & maintaining the list of available events
+* Adding extra info to regular logs (e.g k8s infos)
```
Comment (sgandon) [inline daikon-logging/doc/ECS_migration.adoc:43]:
```suggestion
* Adding extra info to regular logs (e.g k8s infos)
* homogenize the field types and formats
```

File: daikon-logging/doc/ECS_migration.adoc [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+:toclevels: 3
+:toc-placement!:
+
+toc::[]
+
+== Description
+
+Several months ago, https://talend365.sharepoint.com/sites/EnterpriseandFrictionlessInitiativeH1/_layouts/15/Doc.aspx?sourcedoc={14c375d9-ec23-4981-99ac-e061c1241d5b}&action=edit&wd=target%28Reliability.one%7C879313bf-126e-4fe9-b469-80600ece894c%2FLogging%7Cea77767c-3448-4a01-b045-f3db1430968b%2F%29[a discussion] has started in order to define the best way to converge to a common logging format.
+
+The format ECS (Elastic Common Schema) have been chosen to become the common format for all Talend Cloud applications. Each of these applications will have to integrate it.
```
Comment (PierreReliquet) [inline daikon-logging/doc/ECS_migration.adoc:12]:
```suggestion
The format ECS (Elastic Common Schema) has been chosen to become the common format for all Talend Cloud applications. Each of these applications will have to integrate it.
```

**Review by PierreReliquet — COMMENTED** [review]
👍🏻

**Review by sgandon — APPROVED** [review]
LGTM

**Pull request merged**